### ISSUE TITLE
Go type not supported in export: [0]byte

### ISSUE THREAD

**Issue by zoni** [leonrbaker/gomilter#1]
Attempting to build `samplefilter.go` or `samplefilter2.go` results in the following compile error(s) (compiling using `go version go1.2.1 linux/amd64`):

```
# github.com/leonrbaker/gomilter
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
Go type not supported in export: [0]byte
```

I'm not very familiar with Go in general yet and have no experience with cgo at all, but it appears this is an issue with gomilter itself. Some resources that might give extra background info:
- https://groups.google.com/forum/#!topic/golang-nuts/PMippjRxTW0
- https://code.google.com/p/go/issues/detail?id=7409
- https://code.google.com/p/go/source/detail?r=a70a32dc121a52cfdcd20db36e8802c04e8db693

**Comment by Freeaqingme** [comment]
@zoni I'd try upgrading go to 1.4 . I ran across a bug report stating this particular issue and which got fixed somewhere after 1.2. Upgrading to 1.4 fixed this issue for me.

**Comment by leonrbaker** [comment]
Both are compiling in go 1.4
